Build version checker - multiple fixes

jakule opened the pull request:
Export Teleport version in Makefile in build.assets and force `gcc` compiler when running the tool (otherwise, `go` fails to run on a cross-compiled platform).

jakule committed:
Build version checker - multiple fixes

In several files, the command 'go run' has been updated to 'CC=gcc go run'. This ensures that gcc is used when compiling the Go code, to provide better cross platform support. Prior to this change, the Go compiler was making platform specific builds.

This was a significant issue with the 'kube operator' goal where the built binary could not operate across different platforms.

The change has been applied consistently throughout the codebase in 'Makefile', 'container_image_triggers.go', 'version.mk', 'os_repos.go', and 'build.assets/Makefile' files. Compliance with this new standard should be maintained going forward for any new compilation requests.

jakule committed:
Update .drone.yml

jakule commented:
The test build still fails https://drone.platform.teleport.sh/gravitational/teleport/27300 but for a different reason 😞 
I assume that this PR fixes the build issues related to the `version checker`

capnspacehook commented on .drone.yml:
```diff
@@ -17069,6 +17069,6 @@ image_pull_secrets:
 ---
 kind: signature
-hmac: a0cd6e281f864553c31bf642861283cdfa54275907721fcbd1949b3ca239a3e1
+hmac: 54c72179e16cb5f8c2c7bfecdb28ce2bfcbe73c671b555b610a9e353f656abd3
```
Why did this change?

zmb3 replied:
This is a signature of the file. It changes any time there are changes to .drone.yml.

zmb3 commented:
Why do we need to set `CC`? There is no CGo here? Can set set `CGO_ENABLED=0` instead?

tcsc commented:
> Why do we need to set `CC`? There is no CGo here? Can set set `CGO_ENABLED=0` instead?
My thoughts exactly. I am more concerned that the version checker script is trying to do something that requires cgo in the first place.

jakule commented:
> Why do we need to set `CC`? There is no CGo here? Can set set `CGO_ENABLED=0` instead?

My bad, replaced. Fix in 10. Waiting for my build to finish.

jakule commented:
>My thoughts exactly. I am more concerned that the version checker script is trying to do something that requires cgo in the first place.

We enabled CGO in our makefiles, so then when you run something from `make` it will try to enable CGO by default (my understanding).

jakule committed:
Replace CC=gcc environment variable with CGO_ENABLED=0

The CC=gcc environment variable was replaced with CGO_ENABLED=0 as CGO is not needed.

jakule committed:
Update .drone.yml

ravicious commented:
https://github.com/gravitational/teleport/blob/a2a7ce496f5fd41f5d4ca70d3ce485f9ebc38b33/build.assets/Makefile#L13

FWIW, on Windows we don't have egrep available, so this line results in an error. The error is ignored because of how the Windows scripts are set up, but this wasn't immediately obvious for me. See https://drone.platform.teleport.sh/gravitational/teleport/27688/2/4

ravicious commented on build.assets/Makefile:
```diff
@@ -9,6 +9,8 @@ SRCDIR=/go/src/github.com/gravitational/teleport
 # TODO(hugoShaka) remove HELM_PLUGINS with teleport13 buildbox
 DOCKERFLAGS := --rm=true -v "$$(pwd)/../":$(SRCDIR) -v /tmp:/tmp -w $(SRCDIR) -h $(HOSTNAME) -e GOMODCACHE=$(GOMODCACHE) -e HELM_PLUGINS=/home/ci/.local/share/helm/plugins-new
+# Teleport version - some targets require this to be set
+VERSION ?= $(shell egrep ^VERSION ../Makefile | cut -d= -f2)
```
FWIW, on Windows we don't have egrep available, so this line results in an error. The error is ignored because of how the Windows scripts are set up, but this wasn't immediately obvious for me. See https://drone.platform.teleport.sh/gravitational/teleport/27688/2/4
